src/main/java/org/springframework/data/couchbase/repository/query/AbstractCouchbaseQuery.java

```diff
@@ -65,6 +67,8 @@ public AbstractCouchbaseQuery(CouchbaseQueryMethod method, CouchbaseOperations o
 		ExecutableFindByQuery<?> findOp = operations.findByQuery(type);
 		findOp = (ExecutableFindByQuery<?>) (findOp.inScope(method.getScope()).inCollection(method.getCollection()));
 		this.findOperationWithProjection = findOp;
+		this.removeOp = (ExecutableRemoveByQuery<?>) (operations.removeByQuery(type).inScope(method.getScope())
```

Comment:
If I understand the comment correctly - both are fine.   the first one is operations.findByQuery(), the second is operations.removeByQuery().  I'm actually not sure why a cast is even required - because ExecutableRemoveByQuery (eventually) extends the type returned by withCollection() (RemoveByQueryWithOptions).